### PR TITLE
Keep all bytes of one UTF-8 char on same line (no split)

### PR DESCRIPTION
**Pull request by skorobkov**
`mimemail:rfc2047_utf8_encode/1` can splite multibyte UTF-8 character to difrent lines.
Example:
Russian word "Информация", `mimemail:rfc2047_utf8_encode/1` converting it into 2 strings
`=?UTF-8?Q?=D0=98=D0=BD=D1=84=D0=BE=D1=80=D0=BC=D0=B0=D1=86=D0=B8=D1?=`
`=?UTF-8?Q?=8F?=`
Last character "и" (`=D1=8F`) splitted, and dont show correctly in mail-readers.

### CODE REVIEW

**Comment by seriyps** [comment]
AFAIK, `rfc2047_utf8_encode` should produce strings not longer than 76 chars. If your version detects such a split does it move it to next line or broke 76 char limit?

Please, also add unittests!

**Comment by Dinkor** [comment]
Did "\r\n" count in this 76 bytes?

**Comment by Dinkor** [comment]
`mimemail:rfc2047_utf8_encode(unicode:characters_to_binary("€ € € € € 1234 € € € € 123 € € € € € 1234€"))])` produce this strings:
```
=?UTF-8?Q?=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20123?=
 =?UTF-9?Q?4=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20123=20?=
 =?UTF-9?Q?=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20123?=
 =?UTF-9?Q?4=E2=82=AC?=
```
Each encoded line not longer 75 bytes (76 with first space character).

**Comment by mworrell** [comment]
@seriyps Do you think the line length issue is resolved?

**Comment by seriyps** [comment]
@mworrell I guess it is, but still no unittests for this functionality

**Comment by mworrell** [comment]
@skorobkov kan you add your example (with all the €) as an unit test?

**Comment by skorobkov** [comment]
Sorry for delay.
I have added unit test.

File: test/gen_smtp_util_test.erl [hunk 1]
```diff
@@ -82,4 +82,10 @@ rfc822_addresses_roundtrip_test() ->
...
     ok.
 
-    
+rfc2047_utf8_encode_test() ->
+    UnicodeString = unicode:characters_to_binary("€ € € € € 1234 € € € € 123 € € € € € 1234€"),
+    Encoded = "=?UTF-8?Q?=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20123?=\r\n"
+            ++ " =?UTF-8?Q?4=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20123=20?=\r\n"
+            ++ " =?UTF-8?Q?=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20=E2=82=AC=20123?=\r\n"
+            ++ " =?UTF-8?Q?4=E2=82=AC?=",
+    ?assertEqual(mimemail:rfc2047_utf8_encode(UnicodeString), Encoded).
```
Comment (seriyps) [inline test/gen_smtp_util_test.erl]: According to http://erlang.org/doc/apps/eunit/chapter.html#Assert_macros it's recommended to place 'expected' as a first argument and 'actual result' as 2nd of `assertEqual`

**Comment by seriyps** [comment]
Looks good

**Comment by mworrell** [comment]
I also like it. Merging.

@skorobkov Thank you!

**Pull request merged**